### PR TITLE
New backups: Remove the popup and dead code from the Scala side

### PR DESCRIPTION
**Pull request by makingthematrix**
Apart from removed code, there are a few changes around `FirstLaunchAfterLoginFragment` and `AccountManager`.
That's because the order of steps in initialization changed a bit. Previously, the backup was being restored as the first step
in `accountsService.createAccountManager`. We were initializing the user's account already having all data from
the backup, and afterwards we were performing a "cleanup" - resetting certain preferences which triggered
initial sync. Now, the user's account is created empty and only then we restore the backup. The cleanup had to
be moved to afterwards the backup restoration.
#### APK
[Download build #2729](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2729/artifact/build/artifact/wire-dev-PR3013-2729.apk)
[Download build #2732](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2732/artifact/build/artifact/wire-dev-PR3013-2732.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/MainActivity.scala [hunk 1]
```diff
@@ -543,32 +541,6 @@ class MainActivity extends BaseActivity
       .commit
 
   override def onUsernameSet(): Unit = replaceMainFragment(new MainPhoneFragment, MainPhoneFragment.Tag, addToBackStack = false)
-
-  private def showBackUpIncompatibilityDialog() : Future[Unit] = {
-    def showDialog(accentColor: AccentColor): Future[Option[Boolean]] = showWarningDialog(
-      getString(R.string.back_up_incompatibility_title),
-      getString(R.string.back_up_incompatibility_message),
-      R.string.back_up_incompatibility_action_ok,
-      R.string.back_up_incompatibility_action_do_not_show_again,
```
Comment (gizemb) [inline app/src/main/scala/com/waz/zclient/MainActivity.scala:552]: we can remove string resources as well

File: app/src/main/scala/com/waz/zclient/MainActivity.scala [hunk 2]
```diff
@@ -543,32 +541,6 @@ class MainActivity extends BaseActivity
...
-    val prefs = inject[GlobalPreferences]
-
-    for {
-      shouldWarn <- if (BuildConfig.SHOW_BACK_UP_INCOMPATIBILITY_DIALOG) prefs(GlobalPreferences.ShouldWarnBackUpIncompatibility).apply()
-      else Future.successful(false)
-      color <- accentColorController.accentColor.head
-    } yield {
-      if (shouldWarn) {
-        showDialog(color).foreach {
-          case Some(false) => prefs(GlobalPreferences.ShouldWarnBackUpIncompatibility) := false
```
Comment (gizemb) [inline app/src/main/scala/com/waz/zclient/MainActivity.scala:565]: and this preference too

**Review by gizemb — CHANGES_REQUESTED** [review]
did nothing change in the unit tests?

**Review by mejdoo — APPROVED** [review]
same comments of @gizemb 👍

File: app/src/main/scala/com/waz/zclient/MainActivity.scala (hunk 1 above)
Comment (makingthematrix) [reply app/src/main/scala/com/waz/zclient/MainActivity.scala:552]: ok

File: app/src/main/scala/com/waz/zclient/MainActivity.scala (hunk 2 above)
Comment (makingthematrix) [reply app/src/main/scala/com/waz/zclient/MainActivity.scala:565]: ok

**Comment by makingthematrix** [comment]
> did nothing change in the unit tests?

The changes are all in UI. We don't have unit tests for them.

**Review by gizemb — APPROVED** [review]

**Pull request merged**